src/aws-manager.js

```diff
@@ -139,10 +125,172 @@ class AwsManager {
     this.reportAmiUsage = series.amiUsage.reporter(influx);
   }
 
+  async init(){
+    try {
+      let rawJson = await this.spotRequestContainer.read('internal-provisioner-data');
+      let data = JSON.parse(rawJson);
+      // This is mainly for debugging
+      fs.writeFileSync('internal-data.json', data);
+      this.__spotRequestIdCache = data.managedSpotRequests || []; 
+      this.__internalState = data.internallyTrackedSpotRequests || [];
+    } catch (err) {
+      if (err.code !== 'BlobNotFound') {
+        throw err;
+      }
+      this.__spotRequestIdCache = [];
+      this.__internalState = [];
+    }
+  }
+
+  async saveAwsManagerInternalState() {
+    // We use this for both of the current internal caches... let's just use
+    // the same code for both
+    function expire(item) {
+      let diff = Date.now() - item.created;
+      // If we've known about this for more than 5 days, evict it, otherwise
+      // keep it.  5 days is a good amount since we force kill things even if
+      // they're running after 4 days.  This lets us have a full extra day
+      if (diff > 1000 * 60 * 60 * 24 * 5) {
+        return false;
+      }
+      return true;
+    }
+
+    this.__spotRequestIdCache = this.__spotRequestIdCache.filter(expire);
```

Comment:
@jhford, are spot request ids never reused by EC2? I know that machine instance ids are reused...